Modules/TPC/src/Calibration.cxx

```diff
@@ -0,0 +1,89 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   PostProcessingInterface.cxx
+/// \author Thomas Klemenz
+///
+
+// O2 includes
+#include "TPC/Calibration.h"
+#include "TPCBase/Painter.h"
+#include "TPCBase/CDBInterface.h"
+
+// QC includes
+#include "QualityControl/QcInfoLogger.h"
+
+//root includes
+#include "TCanvas.h"
+
+using namespace o2::quality_control::postprocessing;
+
+namespace o2::quality_control_modules::tpc
+{
+
+void Calibration::initialize(Trigger, framework::ServiceRegistry&)
+{
+  mPedestalCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Pedestals"));
+  mPedestalCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Pedestals_1D"));
+  mPedestalCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Pedestals_2D"));
+
+  mNoiseCanvasVec.emplace_back(std::make_unique<TCanvas>("c_Sides_Noise"));
+  mNoiseCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Noise_1D"));
+  mNoiseCanvasVec.emplace_back(std::make_unique<TCanvas>("c_ROCs_Noise_2D"));
+
+  for (auto& canvasPtr : mPedestalCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+
+  for (auto& canvasPtr : mNoiseCanvasVec) {
+    auto canvas = canvasPtr.get();
+    getObjectsManager()->startPublishing(canvas);
+    getObjectsManager()->addMetadata(canvas->GetName(), "custom", "34");
+  }
+}
+
+void Calibration::update(Trigger t, framework::ServiceRegistry&)
+{
+  ILOG(Info, Support) << "Trigger type is: " << t.triggerType << ", the timestamp is " << t.timestamp << ENDM;
+
+  //========Pedestals==========
+  auto& pedestal = o2::tpc::CDBInterface::instance().getPedestals();
+
+  std::vector<TCanvas*> vecPtrPed;
+  for (auto& canv : mPedestalCanvasVec) {
+    vecPtrPed.emplace_back(canv.get());
+  }
+  o2::tpc::painter::makeSummaryCanvases(pedestal, 300, 0, 0, true, &vecPtrPed);
+
+  //==========Noise===========
+  auto& noise = o2::tpc::CDBInterface::instance().getNoise();
+
+  std::vector<TCanvas*> vecPtrNoise;
+  for (auto& canv : mNoiseCanvasVec) {
+    vecPtrNoise.emplace_back(canv.get());
+  }
+  o2::tpc::painter::makeSummaryCanvases(noise, 300, 0, 0, true, &vecPtrNoise);
```

Comment:
Just for the fun of it and to save some copy'n'paste you could try to do
```suggestion
auto toVector = [](auto& input) { 
  std::vector<TCanvas*> output;
  for (auto& in : input) {
    output.emplace_back(in.get());
  }
  return output;
}

  //========Pedestals==========
  auto& pedestal = o2::tpc::CDBInterface::instance().getPedestals();
  o2::tpc::painter::makeSummaryCanvases(pedestal, 300, 0, 0, true, &toVector(mPedestalCanvasVec));

  //==========Noise===========
  auto& noise = o2::tpc::CDBInterface::instance().getNoise();
  o2::tpc::painter::makeSummaryCanvases(noise, 300, 0, 0, true, &toVector(mNoiseCanvasVec));
```

and perhaps even put the lambda above in some helper header you can also include in the cluster thing.

Reply:
Hi @wiechula, I agree on removing the copy&paste. However, the suggested solution is not working. The lambda gives back a temporary instance of output and the makeSummaryCanvases function complains about `taking address of temporary`. If I place line 59 outside the lambda and let it take a reference to the vector as an argument it still fails with the same error.  I am trying around a bit to fix it.

Reply:
Ok I didn't try this, but make sense. You could do
```c++
auto canvPedestal = toVector(mPedestalCanvasVec);
o2::tpc::painter::makeSummaryCanvases(pedestal, 300, 0, 0, true, canvPedestal);
```
without a reference. Should be `std::move` by the compiler, so no big performance issue.

Reply:
That's also (almost) what I found. :)